### PR TITLE
feat(nest): update nest generators to support name and directory format option

### PR DESCRIPTION
**Pull request by Cammisuli**
## Current Behavior

No support for nameAndDirectoryFormat 

## Expected Behavior
Nest generators now support nameAndDirectoryFormat

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by FrozenPandaz — APPROVED** [review]

File: docs/generated/packages/nest/generators/class.json [hunk 1]
```diff
@@ -15,15 +15,20 @@
...
+        "type": "string",
+        "enum": ["as-provided", "derived"]
+      },
       "project": {
         "description": "The Nest project to target.",
         "type": "string",
         "$default": { "$source": "projectName" },
         "alias": "p",
-        "x-prompt": "What Nest project would you like to target?"
+        "x-deprecated": "Provide the `directory` option instead and use the `as-provided` format. The project will be determined from the directory provided. It will be removed in Nx v18."
```
Comment (jaysoo) [inline docs/generated/packages/nest/generators/class.json:28]: Need to also remove `project` from the `required` array in schema.

File: packages/nest/src/generators/utils/normalize-options.ts [hunk 2]
```diff
@@ -5,22 +5,39 @@ import type {
...
-    name: names(options.name).fileName,
-    path: options.directory,
+    flat: true,
+    name: fileName,
     skipFormat: options.skipFormat,
-    sourceRoot,
+    sourceRoot: directory,
   };
 
+  console.log({ normalizedOptions });
```
Comment (jaysoo) [inline packages/nest/src/generators/utils/normalize-options.ts:39]: Remove

File: packages/nest/src/generators/utils/normalize-options.ts [hunk 3]
```diff
@@ -5,22 +5,39 @@ import type {
...
+      callingGenerator,
+      artifactType,
+      name: options.name,
+      directory: options.directory,
+      project: options.project,
+      flat: options.flat,
+      nameAndDirectoryFormat: options.nameAndDirectoryFormat ?? 'derived',
+    });
+
+  console.log({ directory, fileName });
```
Comment (jaysoo) [inline packages/nest/src/generators/utils/normalize-options.ts:29]: Remove

File: packages/nest/src/generators/utils/normalize-options.ts [hunk 4]
```diff
@@ -5,20 +5,31 @@ import type {
...
   options: NestGeneratorOptions
-): NormalizedOptions {
-  const { sourceRoot } = readProjectConfiguration(tree, options.project);
+): Promise<NormalizedOptions> {
+  const { directory, fileName, nameAndDirectoryFormat } =
+    await determineArtifactNameAndDirectoryOptions(tree, {
+      callingGenerator,
+      artifactType,
+      name: options.name,
+      directory: options.directory,
```
Comment (jaysoo) [inline packages/nest/src/generators/utils/normalize-options.ts:21]: Also need to pass `derivedDirectory: options.directory`. The Nest generators are like the React ones where `directory` is relative to project's source root, so `nx g class foo --project test --directory a/b` will generate `test/src/lib/a/b/foo.ts`.

File: packages/nest/src/generators/utils/normalize-options.ts [hunk 5]
```diff
@@ -5,20 +5,31 @@ import type {
...
+      project: options.project,
+      flat: options.flat,
```
Comment (jaysoo) [inline packages/nest/src/generators/utils/normalize-options.ts:23]: Need `flat: options.flat ?? true` so programmatic  usage of generators maintain the same behaviour.

**Comment by jaysoo** [comment]
Two changes needed to maintain same behavior for derived case, but rest LGTM.

File: packages/nest/src/generators/utils/normalize-options.ts (hunk 5 above)
Comment (jaysoo) [reply packages/nest/src/generators/utils/normalize-options.ts:22]: Maybe it's fine

**Pull request merged**